### PR TITLE
initial version of SCIE busDefinition

### PR DESCRIPTION
**Pull request by drom**

### CODE REVIEW

File: specs/sifive.com/SCIE/decoder/0.1.0/SCIEDecoder_rtl.json5 [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        wire: {
+          onMaster: {width: 32, direction: 'out', presence: 'required'},
+          onSlave:  {width: 32, direction: 'in', presence: 'required'}
+        }
+      },
+      unpipelined: {
+        description: 'unpipelined unit to be used',
+        wire: {
+          onMaster: {width: 1, direction: 'out', presence: 'required'},
+          onSlave:  {width: 1, direction: 'in', presence: 'required'}
```
Comment (richardxia) [inline specs/sifive.com/SCIE/decoder/0.1.0/SCIEDecoder_rtl.json5]: Are the directions for both this and the `pipelined` port flipped? The slave should have this as an output signal, correct?

File: specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5 [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    version: '0.1.0',
+    busType: {
+      vendor: 'sifive.com',
+      library: 'core',
+      name: 'SCIEPipelined',
+      version: '0.1.0',
+    },
+    ports: {
+      clock: {
+        description: 'must be used to clock a positive-edge-triggered pipeline register. The functional unit must contain exactly one stage of pipeline register. for two-cycle SCIE only.',
```
Comment (richardxia) [inline specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5]:
```suggestion
        description: 'Must be used to clock a positive-edge-triggered pipeline register. The functional unit must contain exactly one stage of pipeline register. For two-cycle SCIE only.',
```

If we're going to properly capitalizes sentences for part of the description, then we should properly capitalize the whole description.

File: specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5 [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      },
+      rs1: {
+        description: 'rs1 value',
+        wire: {
+          onMaster: {width: 'XLEN', direction: 'out', presence: 'required'},
+          onSlave:  {width: 'XLEN', direction: 'in', presence: 'required'}
+        }
+      },
+      rs2: {
+        description: 'rs1 value',
```
Comment (richardxia) [inline specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5]:
```suggestion
        description: 'rs2 value',
```

File: specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5 [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        wire: {
+          onMaster: {width: 'XLEN', direction: 'out', presence: 'required'},
+          onSlave:  {width: 'XLEN', direction: 'in', presence: 'required'}
+        }
+      },
+      rd: {
+        description: 'rs1 value',
```
Comment (richardxia) [inline specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5]:
```suggestion
        description: 'rd value',
```

File: specs/sifive.com/SCIE/unpipelined/0.1.0/SCIEUnpipelined.json5 [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      },
+      rs1: {
+        description: 'rs1 value',
+        wire: {
+          onMaster: {width: 'XLEN', direction: 'out', presence: 'required'},
+          onSlave:  {width: 'XLEN', direction: 'in', presence: 'required'}
+        }
+      },
+      rs2: {
+        description: 'rs1 value',
```
Comment (richardxia) [inline specs/sifive.com/SCIE/unpipelined/0.1.0/SCIEUnpipelined.json5]:
```suggestion
        description: 'rs2 value',
```

File: specs/sifive.com/SCIE/unpipelined/0.1.0/SCIEUnpipelined.json5 [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        wire: {
+          onMaster: {width: 'XLEN', direction: 'out', presence: 'required'},
+          onSlave:  {width: 'XLEN', direction: 'in', presence: 'required'}
+        }
+      },
+      rd: {
+        description: 'rs1 value',
```
Comment (richardxia) [inline specs/sifive.com/SCIE/unpipelined/0.1.0/SCIEUnpipelined.json5]:
```suggestion
        description: 'rd value',
```

File: specs/sifive.com/SCIE/decoder/0.1.0/SCIEDecoder_rtl.json5 (hunk 1 above)
Comment (drom) [reply specs/sifive.com/SCIE/decoder/0.1.0/SCIEDecoder_rtl.json5]: fixed

File: specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5 (hunk 2 above)
Comment (drom) [reply specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5]: fixed

File: specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5 (hunk 4 above)
Comment (drom) [reply specs/sifive.com/SCIE/pipelined/0.1.0/SCIEPipelined.json5]: fixed

File: specs/sifive.com/SCIE/unpipelined/0.1.0/SCIEUnpipelined.json5 (hunk 6 above)
Comment (drom) [reply specs/sifive.com/SCIE/unpipelined/0.1.0/SCIEUnpipelined.json5]: fixed

File: specs/sifive.com/SCIE/unpipelined/0.1.0/SCIEUnpipelined.json5 (hunk 5 above)
Comment (drom) [reply specs/sifive.com/SCIE/unpipelined/0.1.0/SCIEUnpipelined.json5]: fixed

**Review by richardxia — APPROVED** [review]
LGTM!

**Pull request merged**